src/shogun/io/serialization/BitseryVisitor.h

```diff
@@ -27,6 +27,13 @@ namespace shogun
 				{
 					m_s.boolValue(*v);
 				}
+
+				void on(std::vector<bool>::reference* v) override
+				{
+					bool tmp = *v;
+					m_s.boolValue(tmp);
```

Comment:
Oh vector<bool> is so good :)

Reply:
yea.. there should be std::vector<bool> without optimisation that behaves like the other vectors